ddr/lib/ddr-scanner/dwarf/DwarfScanner.cpp

```diff
@@ -138,14 +144,20 @@ ddr_dw_finish(
 static int
 ddr_dw_init(
 	int           fd,
+	const char* filepath,
 	Dwarf_Handler errhand,
 	Dwarf_Ptr     errarg,
 	Dwarf_Debug  *dbg,
 	Dwarf_Error  *error)
 {
+#if defined(J9ZOS390) && !defined(OMR_EBCDIC) && defined(__open_xl__)
+	char* filename = a2e_string(strdup(filepath));
+	return dwarf_goff_init_with_GOFF_filename(filename, errhand, errarg, 0, dbg, error);
```

Comment:
The API call signature expects a `non const` filepath, even though it does not modify the value. Hence the `strdup` usage. 

Also the new API call expects an EBCDIC filepath, and is converted accordingly.

Reply:
`a2e_string()` should handle `const char *` just fine, so there's no need for `strdup()` which creates (another) leak.
The string returned from `a2e_string()` could be `NULL` and must be freed otherwise.